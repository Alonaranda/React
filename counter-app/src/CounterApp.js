import React, {useState} from 'react';
import PropTypes from 'prop-types'

//__Functional Component__
const CounterApp = ({value}) => {

    //Hook
    //const state = useState('Picoro');
    //console.log(state);
    const [counter, setCounter] = useState(value);

    //Funcion del Click +1
    const handleAdd = (e) => {
        setCounter(counter +1);
    }
    //Funcion Reset
    const reset = (e) => {
        setCounter(value);
    }

    //Funcion -1
    const restar = (e) => {
        setCounter(counter -1);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>Valor de inicio: {value}</h2>
            <h4>{counter}</h4>

            <button onClick={handleAdd}>+1</button>
            <button onClick={reset}>Reset</button>
            <button onClick={restar}>-1</button>
        </>
    );
}
//PropTypes
CounterApp.propTypes = {
    value: PropTypes.number
}
//PropTypes Default
CounterApp.defaultProps = {
    value: 0
}

export default CounterApp;

