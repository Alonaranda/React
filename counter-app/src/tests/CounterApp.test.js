import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import CounterApp from '../CounterApp';
import React from 'react';

describe('Pruebas en CounterApp', () => {
    //Obtener todo el wrapper para las pruebas
    let wrapper = shallow(<CounterApp/>);

    //Inicializar antes de cada prueba
    beforeEach(()=>{
        //console.log('Empieza prueba reiniciada...');
        wrapper = shallow(<CounterApp/>);
    });

    test('Mostrar <CounterApp/> correctamente', () => {
        //const wrapper = shallow(<CounterApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('Mostrar valor por defecto de 100', () => {
        const value = 100;
        const respuesta = `Valor de inicio: ${value}`;
        const wrapper = shallow(
            <CounterApp value={value}/>
        )
        const valorIniciar = wrapper.find('h2').text();
        //console.log(valorIniciar);
        expect(valorIniciar).toBe(respuesta);
    });
    //Click
    test('Debe incrementar con el boton +1', () => {
        //const wrapper = shallow(<CounterApp/>);
        const btn1 = wrapper.find('button').at(0);
        btn1.simulate('click');
        const counter = wrapper.find('h4').text();
        expect(counter).toBe('1');
    });

    test('Debe decrementar con el boton -1', () => {
        //const wrapper = shallow(<CounterApp/>);
        const btn3 = wrapper.find('button').at(2);
        btn3.simulate('click');
        const counter = wrapper.find('h4').text();
        expect(counter).toBe('-1');
    });

    test('Debe resetear al valor inicial', () => {
        const wrapper = shallow(<CounterApp value={105}/>) ;   
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const counter = wrapper.find('h4').text();

        expect(counter).toBe('105');

    });
});

