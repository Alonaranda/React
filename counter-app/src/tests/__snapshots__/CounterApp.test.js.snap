// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Mostrar <CounterApp/> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    Valor de inicio: 
    0
  </h2>
  <h4>
    0
  </h4>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
